### DISCUSSION TITLE
What to do if the fulltextsearch index does not work?

### DISCUSSION THREAD

**Discussion by szaimen** [nextcloud/all-in-one#1709] [Wiki]
When enabling Fulltextsearch in the AIO interface, it will try to do the fulltextsearch index on the next startup of the Nextcloud container. If that fails, it will try to re-do the index every time the container starts. If that should be the case for you, you can fix the problem by doing the following:

1. report the error log to the upstream project: https://github.com/nextcloud/fulltextsearch/issues (make sure that there is not an open issue about this already)
2. Create the marker file manually, so that it does not repeatedly tries to re-do the index:
```
# Run via CLI:
sudo docker exec -it nextcloud-aio-nextcloud bash
```

```
# Now inside the container
touch "$NEXTCLOUD_DATA_DIR/fts-index.done"
exit
```

This should skip the attempt to do the indexing in the future.

**Comment by RudolfAchter** [comment]
In my case indexing seems to restart in an infinite loop. When the container starts it tries to index all files until hitting the memory limit it seems

I try the workaround with creating the `fts-index.done` file and see if index ever gets finished

![image](https://github.com/nextcloud/all-in-one/assets/56109433/ee6bf560-9a7f-447b-a8a7-53acd46dc97c)

**Comment by gbertolottiKS** [comment]
If we want to start again the index generation, removing the "fts-index.done" file is enough?
Thanks

**Reply by szaimen** [reply]
See https://github.com/nextcloud/all-in-one/discussions/2509

**Reply by gbertolottiKS** [reply]
But we don't want to redo all the process from the ground up (we have more than 120GB of group folder data, so to be redone it's a matter of days): we have almost completed the index generation, and we would restart from where we left

**Reply by szaimen** [reply]
okay, then you could also run the command manually. However not sure if this works since I didnt test it.

**Reply by gbertolottiKS** [reply]
Sorry, to what command do you refer?
What are the steps you think we should do to restart manually the generation?
Thanks for your patience

**Reply by szaimen** [reply]
https://github.com/nextcloud/all-in-one/blob/75fdb2848617836da2014244e08916766d4fcc90/Containers/nextcloud/entrypoint.sh#L767

**Reply by gbertolottiKS** [reply]
Thanks; a last question: before launching this command, the containers should be stopped or active?

**Reply by szaimen** [reply]
Active

**Comment by caplam** [comment]
Hello,

I have an instance of nextcloud aio installed on unraid server.
Till now i've not used fulltextsearch. But i installed paperless-ngx and found it useful. So i decided to use fulltextsearch as i could use it with ocr instead of paperless.
That being said all was running fine with install until several weeks ago when clamav started throwing errors. I have not succeeded in making it working again so i simply deactivated it in aio config.
I'm now trying to have fulltextsearch working. So far it's failure.
I have reseted the index and succesfully ran index test but i launch an index it fails with error:
`PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 31457312 bytes) in /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/elastic/transport/src/Transport.php on line 250`

When i launch "sudo -u www-data php occ fulltextsearch:index"
it starts with : 
`Warning: Failed to set memory limit to 0 bytes (Current memory usage is 2097152 bytes) in Unknown on line 0
The current PHP memory limit is below the recommended value of 512MB.`

I saw dozen of topics about memory limit.
Following advice of @szaimen i set up an env var in nextcloud_aio_mastercontainer docker template: NEXTCLOUD_MEMORY_LIMIT
I set it up to 8192M.
I don't know if it's a limit per process or for the whole instance.
It seems to be taken into account:
![Capture d’écran 2024-04-22 à 13 47 29](https://github.com/nextcloud/all-in-one/assets/26901633/ecd78de1-2cea-4ea3-9ffa-a79818e8997f)
So there must be another place to setup memory limit.
On nextcloud web no matter what i search with fulltextsearch (icon in the center of the top bar) nothing is found.
With the unified search (icon on the right of the top bar) i have results but only within document title or path not within document content.
